### DISCUSSION TITLE
Can I send a full MIDI SysEx message as one array?

### DISCUSSION THREAD

**Discussion by rubendax** [djipco/webmidi#309] [Q&A] [answered]
The MIDI System Exclusive implementation for sending and receiving works very well - especially if you are only sending or receiving. However, if you are building a system which does both sending and receiving it would be nice to have the option of sending the **raw** System Exclusive message as one array, including the status byte and the manufacturer ID bytes. 

The current implementation of `WebMidi.outputs[x].sendSysex(manufacturerID, sysexArray)` is easier to use overall, however, having an option to be able to send the full/raw message allows one to implement matching systems for reading incoming SysEx arrays and populating outgoing SysEx arrays. With the current implementation if you specify certain SysEx index locations for certain values, you need to add an offset for your outgoing messages. 

For example: Say that you are using a 3-byte manufacturer ID and you have specified `const SYSEX_INDEX_PRODUCTID = 4;` When reading incoming SysEx messages, the Product ID byte will be found at index 4, However to send a SysEx message back to that same device, the Product ID byte will be placed at index 0 of the `sysexArray` parameter in .sendSysex().

Thanks as always for all of your hard work.

~Edit: I just discovered the `send(status, [data=[]],  [timestamp=0])` method, which definitely gets me closer to what I'm looking for, however, it is still "off by one" from sending the raw MIDI because of the status byte being separate from the `[data=[]]`.~

**Comment by djipco** [comment, marked as answer]
If I understand your needs correctly, the `send()` method is what you are looking for. The method signature you quoted near the end of your message is for version 2. In v3, the method's signature is simply [`send(message, [options])`](https://webmidijs.org/api/classes/Output#send). I believe this is what you are looking for. This method is available on both the [`Output`](https://webmidijs.org/api/classes/Output#send) and [`OutputChannel`](https://webmidijs.org/api/classes/OutputChannel#send) objects.

By the way, thanks for your support! 🙏🏻

**Reply by rubendax** [reply]
Amazing! I have been using V3 for a while now but I was indeed reading the wrong version of the documentation today 😅

Just moved over to this `send()` method and it is exactly what I was looking for. Thank you!

By the way, the other V3 features are excellent as well. Among others, listening for `portschanged` events has been very helpful. Excited for the future updates.
